src/daft-scan/src/glob.rs

```diff
@@ -62,13 +62,15 @@ impl From<Error> for DaftError {
     }
 }
 
+type FileInfoIterator = Box<dyn Iterator<Item = DaftResult<(String, Option<u64>)>>>;
```

Comment:
We actually have the `FileMetaData` struct already, should we re-use that?

Reply:
I suppose we can carry along the unneeded `FileType` field in `FileMetaData` to make this a bit more readable.